lib.js

```diff
@@ -168,3 +171,32 @@ export function storeListResponseToString (res, opts = {}) {
     return res.results.map(({ link }) => link.toString()).join('\n')
   }
 }
+
+/**
+ * Return validated CARLink or undefined
+ * @param {AnyLink} cid
+ **/
+export function asCarLink (cid) {
+  if (cid.version === 1 && cid.code === CAR.codec.code) {
+    return /** @type {CARLink} */ (cid)
+  }
+}
+
+/**
+ * Return validated CARLink type or exit the process with an error code and message
+ * @param {string} cidStr
+ * @param {object} console
+ * @param {(msg: string) => void} console.error
+ **/
+export function parseCarLink (cidStr, { error } = console) {
+  try {
+    const cid = CID.parse(cidStr.trim())
+    const car = asCarLink(cid)
+    if (car === undefined) {
+      error(`Error: ${cidStr} is not a CAR CID`)
```

Comment:
Kinda weird to destructure `error` from console...

I'd probably just have it throw when not a CAR CID and then in the calling code:

```js
let cid
try {
  cid = parseCarLink(cidStr)
} catch {
  console.error(`Error: invalid CAR CID: ${cidStr}`)
  process.exit(1)
}
```

Reply:
yeah it used to be `parseCarLinkOrExit` but that was weird too. Have simplified it again.

```js
export function parseCarLink (cidStr) {
  try {
    return asCarLink(CID.parse(cidStr.trim()))
  } catch {}
}
```

calling code

```js
const shard = parseCarLink(cidStr)
if (!shard) {
  console.error(`Error: ${cidStr} is not a CAR CID`)
  process.exit(1)
}
```